src/main/scala/arithexpr/arithmetic/simplifier/SimplifyProd.scala

```diff
@@ -37,16 +37,16 @@ object SimplifyProd {
                 factorsComeFromProd: Boolean = false,
                 someFactorsComeFromSum: Boolean = true,
                 someFactorsComeFromPow: Boolean = true): ArithExpr with SimplifiedExpr = {
+    var i = 0
```

Comment:
It might make sense to use a comment mention that we do this to avoid boxing, we might otherwise undo it in the future

Reply:
done